syncr_backend/external_interface/distributed_hash_table_util.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+import time
+from collections import OrderedDict
+from typing import List
+from typing import Tuple
+
+from kademlia.network import Server  # type: ignore
+from kademlia.storage import IStorage  # type: ignore
+
+from syncr_backend.constants import TRACKER_DROP_AVAILABILITY_TTL
+
+_node_instance = None
+
+
+def get_dht(
+    bootstrap_ip_port_pair_list: List[Tuple[str, int]],
+    listen_port: int,
+):
+    """
+    returns the node_instance of the dht
+    if no node instance has been created it
+    connects to the distributed hash table
+    if no bootstrap ip port pair list is given, it starts a new dht
+    :param bootstrap_ip_port_pair_list:
+    list of ip port tuples to connect to the dht
+    :param listen_port: port to listen on
+    :return: instance of server
+    """
+    global _node_instance
+    if _node_instance is None:
+        _node_instance = connect_dht(bootstrap_ip_port_pair_list, listen_port)
+
+
+def connect_dht(
+    bootstrap_ip_port_pair_list: List[Tuple[str, int]],
+    listen_port: int,
+) -> Server:
+    """
+    connects to the distributed hash table
+    if no bootstrap ip port pair list is given, it starts a new dht
+    :param bootstrap_ip_port_pair_list:
+    list of ip port tuples to connect to the dht
+    :param listen_port: port to listen on
+    :return: instance of server
+    """
+    node = Server(storage=DropPeerDHTStorage(TRACKER_DROP_AVAILABILITY_TTL))
+    node.listen(listen_port)
+    loop = asyncio.get_event_loop()
+    if len(bootstrap_ip_port_pair_list) > 0:
+        loop.run_until_complete(node.bootstrap(bootstrap_ip_port_pair_list))
+
+    return node
+
+
+class DropPeerDHTStorage(IStorage):
+    def __init__(self, ttl: int=TRACKER_DROP_AVAILABILITY_TTL) -> None:
+        """
+        By default, max age is a week.
+        """
+        self.data = OrderedDict()  # type: OrderedDict
```

Comment:
what's the key and value types for OrderedDict? I think OrderedDict can take type params (ie `OrderedDict[str, str]`).